src/UCTSearch.cpp

```diff
@@ -235,6 +235,39 @@ void UCTSearch::dump_stats(FastState & state, UCTNode & parent) {
     tree_stats(parent);
 }
 
+void UCTSearch::output_analysis(FastState & state, UCTNode & parent) {
+    if (!parent.has_children()) {
+        return;
+    }
+
+    const int color = state.get_to_move();
+    
+    // sort children, put best move on top
+    parent.sort_children(color);
```

Comment:
It don't think it is safe to call this during the search. You need to loop over the children, collect the statistics, and put them in the right order in your local copy.

Have you thought about whether the order the program gives the moves in should be meaningful? I think in the current proposal it cannot, so you can skip the sorting, and avoid most of this complication.

Reply:
You could leave it to the controller/gui to find the best move (or rather the one with most playouts), but it is of course easier if the best move is the first (and what lizzie is getting right now). 

They also get sorted in dump_stats, is that different in some way that makes it safe there?

Reply:
Nevermind, dump_stats is not called during search.

I was about to just delete it, but then I noticed that the loop later in that function needs a sorted list and handling the case where you always output at least two variations (even if the second one has 0 visits) gets a wee bit complicated if it is not sorted (basically you would have to sort manually the same way sort_children does, first on score, then on color AFAICT).

I did also notice however that sort_children gets the mutex before sorting. Isn't that enough to make it safe even while searching?

I've fixed the other things you pointed out, only this remains.